blocks/rich-text/index.js

```diff
@@ -878,17 +878,35 @@ RichText.defaultProps = {
 };
 
 const RichTextContainer = compose( [
-	withBlockEditContext( ( { isSelected } ) => {
+	withInstanceId,
+	withBlockEditContext( ( context, ownProps ) => {
+		// When explicitly set as not selected, do nothing.
+		if ( ownProps.isSelected === false ) {
+			return {};
+		}
+		// When explicitly set as selected or has onFocus prop provided,
+		// use value stored in the context instead.
+		if ( ownProps.isSelected === true || ownProps.onFocus ) {
+			return {
+				isSelected: context.isSelected,
+			};
+		}
+		// Ensures that only one RichText component can be focused.
 		return {
-			isBlockSelected: isSelected,
+			isSelected: context.isSelected && context.focusedElement === ownProps.instanceId,
+			onSetup: () => {
```

Comment:
When does `onSetup` get called? When the user first focuses into the field? I would have thought it'd be at each's initialization, though it doesn't seem like we should want to set the focused element at all until the user actually focuses within (i.e. only `onFocus`).

Reply:
OnSetup gets called when RichText gets mounted (initial render of the editor or when new block gets added). OnFocus is triggered only when a block gets focus but not on the initial render. That’s why we need to maintain two methods.

Reply:
But I don't understand why we need to set the focused element merely by fact the RichText has initialized.